### PR TITLE
Added state to .push

### PR DESCRIPTION
**Pull request by AlexanderKaran**

### CODE REVIEW

File: src/controllers/router-store/index.tsx [hunk 1]
```diff
@@ -157,6 +157,7 @@ const actions: AllRouterActions = {
               ...nextContext,
               location,
               action,
+              state: location.state,
```
Comment (albertogasparin) [inline src/controllers/router-store/index.tsx:160]: Given we are already storing `location`, does it make sense to store `state` here too? Wouldn't it be duplicated?

**Comment by albertogasparin** [comment]
Great start! Would you mind:
- add `state` support also to the other imperative apis `.replace`, `pushTo`, `replaceTo` ?
- add `state` support to `Link` via props `<Link href="..." state={...} />` 
- ensure that it is available and returned in the hooks (eg in `useRouter`) by adding 1/2 tests there too

Thanks!!

File: src/controllers/router-store/index.tsx (hunk 1 above)
Comment (AlexanderKaran) [reply src/controllers/router-store/index.tsx:160]: Good point. I can remove this line. I can either leave it inside the location or abstract it out and so it is easier to access.

**Comment by AlexanderKaran** [comment]
@albertogasparin Added state to `replace`, `pushTo` and `replaceTo`. I also added state to the `Link` and changed it so we are not copying state and just using it from `location`. Due to the state being in location, I am not sure the extra setup and changes to the `useRouter` are needed as the four other functions get tested for a return in state and location is in `useRouter`

**Comment by albertogasparin** [comment]
Looks great! I have one last ask: can you mention it in the docs? So it's clear that those APIs and Link accept `state` and you can find the value inside `location`

**Comment by AlexanderKaran** [comment]
@albertogasparin Added an example and updated Link props, let me know if you need more

**Review by albertogasparin — APPROVED** [review]

**Pull request merged**